### PR TITLE
Add tooltips

### PR DESCRIPTION
**Pull request by laptou**
- Chart series now have tooltips
- Chart can scroll sideways

https://drive.google.com/file/d/1-jvZfTZHDy7UctOIMnTV0Q1SgnBdk-pA/view?usp=sharing

### CODE REVIEW

File: client/js/chart/index.js [hunk 1]
```diff
@@ -79,7 +92,12 @@ export function createChart(el, rankingHistory, rankingDates) {
       const seriesElement = seriesElements[hit];
       seriesElement.dispatchEvent(
           new CustomEvent('series-hit', {
-            detail: {x, y, key: seriesEntry[0]},
+            detail: {
+              x,
+              y,
+              key: seriesEntry[0],
+            },
+            bubbles: true,
```
Comment (gracetian6) [inline client/js/chart/index.js:100]: formatting like this is more readable:
```
detail: {x, y, key: seriesEntry[0]},
bubbles: true, 
```

File: client/js/chart/index.js [hunk 2]
```diff
@@ -219,3 +239,44 @@ function createDefs() {
...
+    // convert SVG coords to HTML coords
+    let pos = svg.createSVGPoint();
+    pos.x = x * RUN_SCALE_X;
+    pos.y = y * RUN_SCALE_Y;
+    pos = pos.matrixTransform(svg.getScreenCTM());
+    pos.x -= svgBounds.x;
+    pos.y -= svgBounds.y;
+
+    tooltip.classList.add('chart-tooltip-active');
+    tooltip.style.transform = `translate(${pos.x}px, ${pos.y}px)`;
```
Comment (gracetian6) [inline client/js/chart/index.js:271]: bougie $$ 🤑

File: client/js/chart/index.js [hunk 3]
```diff
@@ -219,3 +239,44 @@ function createDefs() {
...
+/**
+ * Creates a tooltip that responds to the user pointing at things on the chart.
+ *
+ * @param {SVGSVGElement} svg The SVG element for the chart.
+ * @param {Map<string, number[]>} rankingHistory The ranking history for each
+ * track.
+ * @param rankingDates
+ * @returns {HTMLDivElement} An element for the tooltip.
+ */
+function createTooltip(svg, rankingHistory, rankingDates) {
```
Comment (gracetian6) [inline client/js/chart/index.js:252]: how is creating a tooltip different from hit test?

Comment (laptou) [reply client/js/chart/index.js:252]: Hit test determines where the tooltip should go

File: client/js/chart/index.js (hunk 1 above)
Comment (laptou) [reply client/js/chart/index.js:100]: Why?

**Pull request merged**